### PR TITLE
modularize road class

### PR DESCRIPTION
**Pull request by rajnikantsh**
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/2241)

### CODE REVIEW

**Comment by JustinPihony** [comment]
Is this still valid for merging?

File: src/main/java/beam/physsim/jdeqsim/cacc/sim/Road.java [hunk 1]
```diff
@@ -224,127 +177,50 @@ private double  getNextAvailableTimeForLeavingStreet(double simTime){
...
-            double minTimeForNextDeadlockPreventionMessageTime=0;
+        } else {
+            nextStuckTime=simTime + config.getSqueezeTime();
+        }
 
-            if (deadlockPreventionMessages_.size() > 0) minTimeForNextDeadlockPreventionMessageTime=deadlockPreventionMessages_.getLast().getMessageArrivalTime()+0.0000000001; // ensures that deadlock prevention messages have increasing time stamps - this is assumped by original implementation around this
+        if (!Road.getRoad(vehicle.getCurrentLinkId()).latestTimeToLeaveRoad.containsKey(vehicle)){
+            Road.getRoad(vehicle.getCurrentLinkId()).latestTimeToLeaveRoad.put(vehicle,simTime + link.getLength()/minimumRoadSpeedInMetersPerSecond);
+        } else {
+            System.out.print("");
```
Comment (REASY) [inline src/main/java/beam/physsim/jdeqsim/cacc/sim/Road.java]: Let's get ride of  this or if prints. If some log message is required, let add proper logging

Comment (rajnikantsh) [reply src/main/java/beam/physsim/jdeqsim/cacc/sim/Road.java]: removed this print statement

File: src/main/java/beam/physsim/jdeqsim/cacc/sim/Road.java [hunk 2]
```diff
@@ -121,93 +112,55 @@ private double getInverseCapacity(org.matsim.core.mobsim.jdeqsim.Vehicle vehicle
...
-        gapTravelTime_ =(Double)getField(this,"gapTravelTime");
-        inverseInFlowCapacity_ =(Double)getField(this,"inverseInFlowCapacity");
+        nextVehicle.scheduleEnterRoadMessage(nextAvailableTimeForEnteringStreet, this);
+        latestTimeToLeaveRoad.remove(vehicle);
+    }
 
+    public void leaveRoad(org.matsim.core.mobsim.jdeqsim.Vehicle vehicle, double simTime) {
         assert (this.carsOnTheRoad.getFirst() == vehicle);
-        assert (this.interestedInEnteringRoad_.size()==this.deadlockPreventionMessages_.size());
+        assert (this.getInterestedInEnteringRoad().size()==this.getInterestedInEnteringRoad().size());
```
[the remark this answers is not in the document]
Comment (rajnikantsh) [reply src/main/java/beam/physsim/jdeqsim/cacc/sim/Road.java]: change this comparision with deadlockPreventionMessages

**Pull request merged**